### PR TITLE
Add random string utility 

### PR DESCRIPTION
**Pull request by abdulhaseeb2**

### CODE REVIEW

**Comment by stakater-user** [comment]
@abdulhaseeb2 Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/jira-service-desk-operator/commit/4083c42408ecf4daca7f45299f5af2a72c85eacd/checks) has Failed!

**Comment by stakater-user** [comment]
@abdulhaseeb2 Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/jira-service-desk-operator/commit/a661c16d10ee8f7ededccbf0f534889478653001/checks) has Failed!

**Comment by stakater-user** [comment]
@abdulhaseeb2 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-56-9277aeba`

**Comment by stakater-user** [comment]
@abdulhaseeb2 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-56-0ac4100e`

**Comment by stakater-user** [comment]
@abdulhaseeb2 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-56-112ba23d`

File: controllers/customer_controller_test.go [hunk 1]
```diff
@@ -24,6 +25,9 @@ var _ = Describe("Customer Controller", func() {
 	Describe("Create new Jira Service Desk customer", func() {
 		Context("With valid fields", func() {
 			It("should create a new customer", func() {
+				// Randomize customer name and email
+				mockData.SampleCustomer = cUtil.RandCustomer(mockData.SampleCustomer)
```
Comment (rehanjaved237-zz) [inline controllers/customer_controller_test.go]: This random customer should be created outside the describe block. In current case, the after each block won't be able to delete it after this test case has run

File: controllers/customer_controller_test.go [hunk 2]
```diff
@@ -39,14 +43,17 @@ var _ = Describe("Customer Controller", func() {
 					project := util.GetProject(mockData.CustomerTestProjectInput.Spec.Name, ns)
 					Expect(project.Status.ID).ToNot(Equal(""))
 
+					// Randomize customer name and email
+					mockData.SampleCustomer = cUtil.RandCustomer(mockData.SampleCustomer)
```
Comment (rehanjaved237-zz) [inline controllers/customer_controller_test.go]: Same as above. This should be outside. In the current scenario, the aftereach block won't be able to handle it

File: controllers/customer_controller_test.go [hunk 3]
```diff
@@ -62,6 +69,11 @@ var _ = Describe("Customer Controller", func() {
 					project := util.GetProject(mockData.CustomerTestProjectInput.Spec.Name, ns)
 					Expect(project.Status.ID).ToNot(Equal(""))
 
+					// Randomize customer name and email
+					mockData.SampleCustomer = cUtil.RandCustomer(mockData.SampleUpdatedCustomer)
```
Comment (rehanjaved237-zz) [inline controllers/customer_controller_test.go]: Customer should be declared outside.

File: controllers/customer_controller_test.go [hunk 4]
```diff
@@ -83,6 +95,7 @@ var _ = Describe("Customer Controller", func() {
 	Describe("Delete Jira Service Desk customer", func() {
 		Context("With valid Customer AccountId", func() {
 			It("should delete the customer", func() {
+
 				_ = cUtil.CreateCustomer(mockData.SampleCustomer, ns)
```
Comment (rehanjaved237-zz) [inline controllers/customer_controller_test.go]: Customer should be declared outside of this block

File: controllers/util/testUtil.go [hunk 5]
```diff
@@ -30,6 +32,25 @@ func New(ctx context.Context, k8sClient client.Client, r reconcile.Reconciler) *
 	}
 }
 
+// RandCustomer Generates random customer names and emails
+func (t *TestUtil) RandCustomer(customer jiraservicedeskv1alpha1.Customer) jiraservicedeskv1alpha1.Customer {
```
Comment (rehanjaved237-zz) [inline controllers/util/testUtil.go]: Comment should be something like RandCustomer updates a customer with a random name and email

File: controllers/project_controller_test.go [hunk 6]
```diff
@@ -115,6 +122,9 @@ var _ = Describe("Project Controller", func() {
 		Describe("Create new Jira servie desk project resource", func() {
 			Context("with invalid fields", func() {
 				It("should not create a new project", func() {
+					key := cUtil.RandSeqString(10)
+					projectInvalidInput.Spec.Key = strings.ToUpper(key)
```
Comment (rehanjaved237-zz) [inline controllers/project_controller_test.go:126]: Key can not be of 10 digits. I think you replaced the value for key with name.

**Review by rehanjaved237-zz — CHANGES_REQUESTED** [review]

File: controllers/util/testUtil.go (hunk 5 above)
Comment (abdulhaseeb2) [reply controllers/util/testUtil.go]: ack

File: controllers/project_controller_test.go (hunk 6 above)
Comment (abdulhaseeb2) [reply controllers/project_controller_test.go:126]: I changed key size to be 9 and projectInvalidInput.Spec.Name += key[:3] this line just appends a random 3 char long string to existing name.

File: controllers/customer_controller_test.go (hunk 4 above)
Comment (abdulhaseeb2) [reply controllers/customer_controller_test.go]: I changed nothing in this code block

**Comment by stakater-user** [comment]
@abdulhaseeb2 Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/jira-service-desk-operator/commit/12c61a3d835550f2e3c6c011bfeda473a4099c80/checks) has Failed!

**Comment by stakater-user** [comment]
@abdulhaseeb2 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-56-22c1a1e5`

File: controllers/suite_test.go [hunk 7]
```diff
@@ -129,6 +131,12 @@ var _ = BeforeSuite(func(done Done) {
 	cUtil = controllerUtil.New(ctx, k8sClient, cr)
 	Expect(util).ToNot(BeNil())
 
+	// Generation of 3 char long random string
+	cusKey = cUtil.RandSeqString(3)
+
+	mockData.CustomerTestProjectInput.Spec.Name += cusKey
+	mockData.CustomerTestProjectInput.Spec.Key = strings.ToUpper(cusKey)
+
 	_ = util.CreateProject(mockData.CustomerTestProjectInput, ns)
```
Comment (rehanjaved237-zz) [inline controllers/suite_test.go:137]: This project created here is not deleted in AfterSuite.

**Review by rehanjaved237-zz — CHANGES_REQUESTED** [review]

File: controllers/suite_test.go (hunk 7 above)
Comment (abdulhaseeb2) [reply controllers/suite_test.go:137]: Ack

**Comment by stakater-user** [comment]
@abdulhaseeb2 Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/jira-service-desk-operator/commit/c0a5c45c88ae0feffb09fc7ededfb74fbaaf8b73/checks) has Failed!

**Comment by stakater-user** [comment]
@abdulhaseeb2 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-56-4f803773`

**Comment by stakater-user** [comment]
@abdulhaseeb2 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-56-5d59a606`

**Comment by stakater-user** [comment]
@abdulhaseeb2 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-56-2fe09854`

**Review by rehanjaved237-zz — DISMISSED** [review]
LGTM

File: controllers/util/testUtil.go [hunk 8]
```diff
@@ -30,6 +32,25 @@ func New(ctx context.Context, k8sClient client.Client, r reconcile.Reconciler) *
 	}
 }
 
+// RandCustomer updates a customer with a random name and email
```
Comment (ahmedwaleedmalik) [inline controllers/util/testUtil.go]: Remove this method and make the conversion explicit, where it is requried.

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]
Left comment

File: controllers/util/testUtil.go (hunk 8 above)
Comment (abdulhaseeb2) [reply controllers/util/testUtil.go]: ack

**Comment by stakater-user** [comment]
@abdulhaseeb2 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-56-3b697f8b`

**Review by ahmedwaleedmalik — APPROVED** [review]
LGTM

**Pull request merged**